### PR TITLE
Remove all usages of `std.mem.copy` and remove `std.mem.set`

### PR DESCRIPTION
**Pull request by Rexicon226**
My reasoning for making this PR is 2 reasons.

1. This should have been simply removed/`@compileError` the first time, and now there is just no point in keeping it around.

2. In my opinion this creates a bad practice function alias. By aliasing `copyForward`, we are creating possible bugs that are not caught and corrupt data due to overlap. The pointer overlap detection of `@memcpy` is more important than I believe given credit to. So it leads to the thought that all uses of `std.mem.copy` (after it was aliased/depricated), were made out of ignorance of the dangerous it could pose to the codebase. If the writer wanted to ensure overlap, they would have / should use `copyForward`, and if they did *not* want overlap, they need to use `@memcpy`.

### CODE REVIEW

**Comment by squeek502** [comment]
Just FYI, the typical procedure is to alias with a `Deprecated` doc comment for one release cycle and then remove the alias during the next release cycle, so this is fair game since the deprecation note was included in `0.11.0`.

File: src/resinator/source_mapping.zig [hunk 1]
```diff
@@ -476,7 +476,7 @@ pub const SourceMappings = struct {
 
         const after_collapsed_start = line_num + num_following_lines_to_collapse;
         const new_num_lines = self.mapping.items.len - num_following_lines_to_collapse;
-        std.mem.copy(SourceSpan, self.mapping.items[line_num..new_num_lines], self.mapping.items[after_collapsed_start..]);
+        @memcpy(self.mapping.items[line_num..new_num_lines], self.mapping.items[after_collapsed_start..]);
```
[the remark this answers is not in the document]
Comment (Rexicon226) [reply src/resinator/source_mapping.zig:479]: maybe... the boundaries are defined very precisely though, so I don't know.

**Comment by Rexicon226** [comment]
For the CI failures, should I just change them all to `copyForward`? Or is there something more indepth that I should do?

**Comment by squeek502** [comment]
The slices need to be the same length to use `@memcpy`, see https://github.com/ziglang/zig/pull/15481 for how things were updated when the change was made.

**Comment by Rexicon226** [comment]
ah i see, they were just done using `[0..source.len]` inside of the `@memcpy`. I'll do that rn!

**Comment by Rexicon226** [comment]
i know what to fix, i just wanna test a couple things tomorrow :smile:

i wish there was a way for me to cancel ci actions, because i feel bad about wasting resources :sob:

**Review by Vexu — APPROVED** [review]
Please submit a patch upstream as well!

**Pull request merged**